files/lib/system/conversation/ConversationHandler.class.php

```diff
@@ -122,4 +123,33 @@ public function getConversationCount($userID = null) {
 		
 		return $this->conversationCount[$userID];
 	}
+	
+	/**
+	 * Enforces the flood control.
+	 */
+	public function enforceFloodControl() {
+		$limit = WCF::getSession()->getPermission('user.conversation.maxStartedConversationsPer24Hours');
+		if ($limit == -1) {
+			return;
+		}
+		
+		$sql = "SELECT          COUNT(*) AS count, MIN(time) AS oldestDate
+			FROM            wcf" . WCF_N . "_conversation
+			WHERE           userID = ?
+					AND time > ?
+			GROUP BY        userID";
```

Comment:
You can get rid of the `GROUP BY userID`, because you'll only ever have a single `userID`. If you do then a configuration of `0` will result in (which is what I assumed to happen anyway):

```
+-------+------------+
| count | oldestDate |
+-------+------------+
|     0 |       NULL |
+-------+------------+
```